assignment/src/com/assignmentweek2/BinarySearch/Main.java

```diff
@@ -0,0 +1,51 @@
+package com.assignmentweek2.BinarySearch;
+
+import java.util.Scanner;
+import java.util.TreeSet;
+/*
+This is the main method
+ */
+public class Main {
+    public static void main(String[] args) {
+        TreeSet <Number> elementSet =new TreeSet<>();
```

Comment:
Use generics to write common logic for search and sort.

Reply:
changed